sycl/doc/extensions/SubGroup/SYCL_INTEL_sub_group.asciidoc

```diff
@@ -0,0 +1,251 @@
+
+|+info::device::sub_group_independent_forward_progress+
+|+bool+
+|Returns +true+ if the device supports independent forward progress of sub-groups with respect to other sub-groups in the same work-group.
+
+|+info::device::sub_group_sizes+
+|+vector_class<size_t>+
+|Returns a vector_class of +size_t+ containing the set of sub-group sizes supported by the device.
+|===
+
+|===
+|Member Functions|Description
+
+|+template <info::kernel_sub_group param>typename info::param_traits<info::kernel_sub_group, param>::return_type get_sub_group_info(const device &dev) const+
+|Query information from the sub-group from a kernel using the +info::kernel_sub_group+ descriptor for a specific device.
+|===
+
+|===
+|Kernel Descriptors|Return Type|Description
+
+|+info::kernel_sub_group::max_num_sub_groups+
+|+uint32_t+
+|Returns the maximum number of sub-groups for this kernel.
+
+|+info::kernel_sub_group::compile_num_sub_groups+
+|+uint32_t+
+|Returns the number of sub-groups specified by the kernel, or 0 (if not specified).
+
+|+info::kernel_sub_group::max_sub_group_size+
+|+uint32_t+
+|Returns the maximum sub-group size for this kernel.
+
+|+info::kernel_sub_group::compile_sub_group_size+
+|+uint32_t+
+|Returns the required sub-group size specified by the kernel, or 0 (if not specified).
+|===
+
+=== The sub_group Class
+
+The +sub_group+ class encapsulates all functionality required to represent a particular sub-group within a parallel execution.  It has common by-value semantics and is not default or user-constructible, and can only be accessed via methods in the +nd_item+ class.
+
+|===
+|Member Functions|Description
+
+|+sub_group get_sub_group() const+
+|Return the sub-group to which the work-item belongs.
+|===
+
+The member functions of the sub-group class provide a mechanism for a developer to query properties of a sub-group and a work-item's position in it.
+
+|===
+|Member Functions|Description
+
+|+id<1> get_local_id() const+
+|Return an +id+ representing the index of the work-item within the sub-group.
+
+|+uint32_t get_linear_local_id() const+
+|Return a +uint32_t+ representing the index of the work-item within the sub-group.
+
+|+range<1> get_local_range() const+
+|Return a SYCL +range+ representing the number of work-items in the sub-group.
+
+|+range<1> get_max_local_range() const+
+|Return a SYCL +range+ representing the maximum number of work-items in any sub-group within the nd-range.
+
+|+id<1> get_group_id() const+
+|Return an +id+ representing the index of the sub-group within the work-group.
+
+|+uint32_t get_linear_group_id() const+
+|Return a +uint32_t+ representing the index of the sub-group within the work-group.
+
+|+range<1> get_group_range() const+
+|Return a SYCL +range+ representing the number of sub-groups within the work-group.
+
+|+range<1> get_uniform_group_range() const+
+|Return a SYCL +range+ representing the number of sub-groups per work-group in the uniform region of the nd-range.
```

Comment:
Since SYCL spec doesn't support non-uniform work-groups, `uniform region of the nd-range` seems unspecified to me.

Reply:
I agree.  Looking at this again, I'm not sure that the description matches the name or what I would expect the function to do, suggesting that there might be some sort of copy-paste error at work here.

Reply:
Looks like it was intended to ensure that sub-groups were compatible with non-uniform work-groups, even though they're not supported by SYCL 1.2.1.  I've not seen anybody using this, so I'll remove it for now.